src/AutoMapper/Bundle/AutoMapper.php

```diff
@@ -3,24 +3,14 @@
 namespace Jane\AutoMapper\Bundle;
 
 use Jane\AutoMapper\AutoMapper as BaseAutoMapper;
-use Jane\AutoMapper\Bundle\Configuration\ConfigurationPassInterface;
-use Jane\AutoMapper\MapperGeneratorMetadataInterface;
+use Jane\AutoMapper\Bundle\Configuration\MapperConfigurationInterface;
 
 class AutoMapper extends BaseAutoMapper
 {
-    /** @var ConfigurationPassInterface[] */
-    private $configurationPass = [];
-
-    public function addConfigurationPass(ConfigurationPassInterface $configurationPass): void
-    {
-        $this->configurationPass[] = $configurationPass;
-    }
-
-    public function register(MapperGeneratorMetadataInterface $metadata): void
+    public function addMapperConfiguration(MapperConfigurationInterface $mapperConfiguration): void
     {
-        foreach ($this->configurationPass as $configurationPass) {
-            $configurationPass->process($metadata);
-        }
+        $metadata = $this->getMetadata($mapperConfiguration->getSource(), $mapperConfiguration->getTarget());
+        $mapperConfiguration->process($metadata);
 
         parent::register($metadata);
```

Comment:
why calling `parent::register($metadata)` each time we add a Mapper configuration ?

Reply:
You're right, it's definitively not needed! :ok_hand: